### PR TITLE
auto restart service if we note it's running an older version

### PR DESCRIPTION
**Pull request by maxtaco**
- mainInner was a real mess, let's clean it up
- add a KEYBASE_SET_VERSION environment variable for testing
- everything is tested under autofork and brew modes

Closes CORE-1989

### CODE REVIEW

File: go/client/versionfix.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	if err != nil {
+		return err
+	}
+	semverService, err = semver.Make(serviceConfig.Version)
+	if err != nil {
+		return err
+	}
+
+	g.Log.Debug("| version check %s v %s", semverClient, semverService)
+	if !semverClient.GT(semverService) {
```
Comment (patrickxb) [inline go/client/versionfix.go]: is there a "not equal" func we could use instead?  In case they use an old client binary somehow?

**Comment by patrickxb** [comment]
LGTM after changing FixVersionClash to work w/ any version clash, not just newer client version.  Also, appveyor CI issues...

**Comment by maxtaco** [comment]
thanks for the comments... it looks like windows CI is totally borked.  I can't imagine my change did this, I wonder if it's bad on the mainline too?  It could be a ci1 failure.

File: go/client/versionfix.go (hunk 1 above)
Comment (maxtaco) [inline go/client/versionfix.go]: But then a restart wouldn't fix anything.  The same condition would apply the next time.  Better to just leave it alone, no?

Comment (maxtaco) [inline go/client/versionfix.go]: Changed it to error out, is this ok?

**Pull request merged**